gov2/rds/actions/instances.go

```diff
@@ -0,0 +1,294 @@
+		return nil, err
+	} else {
+		return output.DBSnapshot, nil
+	}
+}
+
+// snippet-end:[gov2.rds.CreateDBSnapshot]
+
+// snippet-start:[gov2.rds.DescribeDBSnapshots]
+
+// GetSnapshot gets a DB instance snapshot.
+func (instances *DbInstances) GetSnapshot(snapshotName string) (*types.DBSnapshot, error) {
+	output, err := instances.RdsClient.DescribeDBSnapshots(context.TODO(),
+		&rds.DescribeDBSnapshotsInput{
+			DBSnapshotIdentifier: aws.String(snapshotName),
+		})
+	if err != nil {
+		log.Printf("Couldn't get snapshot %v: %v\n", snapshotName, err)
+		return nil, err
+	} else {
+		return &output.DBSnapshots[0], nil
+	}
+}
+
+// snippet-end:[gov2.rds.DescribeDBSnapshots]
+
+// snippet-start:[gov2.rds.CreateDBInstance]
+
+// CreateInstance creates a DB instance.
+func (instances *DbInstances) CreateInstance(instanceName string, dbName string,
+	dbEngine string, dbEngineVersion string, parameterGroupName string, dbInstanceClass string,
+	storageType string, allocatedStorage int32, adminName string, adminPassword string) (
+	*types.DBInstance, error) {
+	output, err := instances.RdsClient.CreateDBInstance(context.TODO(), &rds.CreateDBInstanceInput{
+		DBInstanceIdentifier: aws.String(instanceName),
+		DBName:               aws.String(dbName),
+		DBParameterGroupName: aws.String(parameterGroupName),
+		Engine:               aws.String(dbEngine),
+		EngineVersion:        aws.String(dbEngineVersion),
+		DBInstanceClass:      aws.String(dbInstanceClass),
+		StorageType:          aws.String(storageType),
+		AllocatedStorage:     aws.Int32(allocatedStorage),
+		MasterUsername:       aws.String(adminName),
+		MasterUserPassword:   aws.String(adminPassword),
+	})
+	if err != nil {
+		log.Printf("Couldn't create instance %v: %v\n", instanceName, err)
+		return nil, err
+	} else {
+		return output.DBInstance, nil
+	}
+}
+
+// snippet-end:[gov2.rds.CreateDBInstance]
+
+// snippet-start:[gov2.rds.DescribeDBInstances]
+
+// GetInstance gets data about a DB instance.
+func (instances *DbInstances) GetInstance(instanceName string) (
+	*types.DBInstance, error) {
+	output, err := instances.RdsClient.DescribeDBInstances(context.TODO(),
+		&rds.DescribeDBInstancesInput{
+			DBInstanceIdentifier: aws.String(instanceName),
+		})
+	if err != nil {
+		var notFoundError *types.DBInstanceNotFoundFault
+		if errors.As(err, &notFoundError) {
+			log.Printf("DB instance %v does not exist.\n", instanceName)
+			err = nil
+		} else {
+			log.Printf("Couldn't get instance %v: %v\n", instanceName, err)
+		}
+		return nil, err
+	} else {
+		return &output.DBInstances[0], nil
+	}
+}
+
+// snippet-end:[gov2.rds.DescribeDBInstances]
+
+// snippet-start:[gov2.rds.DeleteDBInstance]
+
+// DeleteInstance deletes a DB instance.
+func (instances *DbInstances) DeleteInstance(instanceName string) error {
+	_, err := instances.RdsClient.DeleteDBInstance(context.TODO(), &rds.DeleteDBInstanceInput{
+		DBInstanceIdentifier:   aws.String(instanceName),
+		SkipFinalSnapshot:      true,
```

Comment:
Does this parameter need to be wrapped like `aws.Bool(true)` ?

Reply:
No, this is a quirk of the Go SDK. I don't know why, but sometimes they want pointers and other times not. This is a case where non-pointer is needed.